### PR TITLE
Define Same Domain

### PR DESCRIPTION
**Pull request by tomrittervg**

### CODE REVIEW

File: draft-ietf-trans-gossip.md [hunk 1]
```diff
@@ -249,25 +249,30 @@ described in {{feedback-srvaud}}.
...
 the remaining SCTs together with the corresponding certificate chain
 for later use in SCT Feedback.
 
-When the client later reconnects to any HTTPS server for the same
-domain, it again receives a set of SCTs. The client MUST add new SCTs
-from known logs to its store of SCTs for the server. The client MUST
-send to the server any SCTs in the store that are associated with that
-server but which were not received from that server.
+The SCTs stored on the client MUST be keyed by the exact domain name 
+the client contacted. They MUST NOT be sent to any domain related
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:258]: Missing a "not" here? 

> They MUST NOT be sent to any domain not related

File: draft-ietf-trans-gossip.md [hunk 2]
```diff
@@ -249,25 +249,30 @@ described in {{feedback-srvaud}}.
...
+NOT be sent to any Subject Alternate Names specified in the certificate.
+Disobeying these security requirements will leak user's browing habits 
+cross-domain. In the case of certificates that validate multiple domain
+names, it is expected that the same SCT may be stored multiple times.
+
+When the client later connects to an HTTPS server it again receives a 
+set of SCTs. The client MUST add new SCTs from known logs to its store 
+of SCTs for the individual domain name contacted. The client MUST
+send to the server SCTs in the store that are associated with that
+domain name but which were not received from that server.
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:271]: Can you elaborate on the last sentence, especially the last part starting with "but which ..."?

File: draft-ietf-trans-gossip.md (hunk 1 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:258]: Yes, you're correct I am missing that lowercase not.

File: draft-ietf-trans-gossip.md [hunk 3]
```diff
@@ -280,14 +285,18 @@ be unrevoked (which will be hard for a MitM to sustain)\]
...
+is an exact match for the domain the SCT was originally received under. Not
 following this constraint would lead to two types of privacy
 leaks. First, the server receiving the SCT would learn about other
-sites visited by the HTTPS client. Secondly, auditors or monitors
+sites visited by the HTTPS client, as explained above. Secondly, auditors or monitors
 receiving SCTs from the HTTPS server would learn information about the
 other HTTPS servers visited by its clients.
 
+\[TODO: this MUST is a repeat of the above. Do we need to re-iterate this? Probably not, so we can remove this paragraph? \]
+
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:299]: Move the rationale (starting with "Not following this...") upwards and remove the MUST?

File: draft-ietf-trans-gossip.md (hunk 2 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:271]: Ah.  It's saying "Don't bother sending back the exact same SCTs you just received."

File: draft-ietf-trans-gossip.md (hunk 3 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:299]: Agree.

File: draft-ietf-trans-gossip.md (hunk 1 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:258]: Fixed in tjr-define-same-domain in my repo.

File: draft-ietf-trans-gossip.md (hunk 2 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:271]: OK, I'll try rewording it to make that clear, and also to make clear that "just received" means "in this very same TLS connection". Sending SCT:s back to what the client think is the same server but might not be the same server is kind of crucial.

File: draft-ietf-trans-gossip.md (hunk 3 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:299]: I'll fix this.

**Comment by ln5** [comment]
Take a look at https://github.com/ln5/draft-ietf-trans-gossip/tree/tjr-define-same-domain and let me know if that reflects what you wanted to say in your pull request.

**Comment by tomrittervg** [comment]
Looks good!  I noticed one typo I made: exmaple

**Comment by ln5** [comment]
Fixed and merged to master.

But also, heh:

> % dig +short exmaple.com ns
> b.authns.bitfolk.com.
> a.authns.bitfolk.co.uk.
> numpty.absolutelyplastered.com.
> c.authns.bitfolk.com.

**Pull request merged**